baseapp/params.go

```diff
@@ -47,6 +48,7 @@ func DefaultParams(logger zerolog.Logger, metricsPrefix string) []Param {
 		WithUTCNanoTime(),
 		WithErrorLogging(RichErrorMarshalFunc),
 		WithMetrics(),
+		WithTLSConfig(&tls.Config{MinVersion: tls.VersionTLS12}),
```

Comment:
Do we need to add MinVersion to the [TLSConfig struct](https://github.com/palantir/go-baseapp/blob/develop/baseapp/config.go#L34)? Perhaps adding these options set to their defaults in the [`example/config.yaml`](https://github.com/palantir/go-baseapp/blob/develop/example/config.yml#L1) may help with discoverability as well?

Reply:
hrmm. i guess it depends on how we want to treat TLS configuration: is it runtime configurable, specifically: will a single binary be deployed to different scenarios where we want to set the minimum TLS version.

considering TLS 1.0 and 1.1 have been widely deprecated since early 2020, i feel okay with increasing the security for anyone who keeps the defaults at the cost of downstream consumers that require TLS 1.0/1.1 support to explicitly add code to configure their `MinVersion` in their services.

Reply:
I think we should leave this as a code-only option until there are two or more places where having it as part of the configuration file would be useful.